rfcs/20210510-sum-with-zero-point/README.md

```diff
@@ -0,0 +1,158 @@
+            else
+                if (zero_point == 0)
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v2(
+                                              get(), scale,
+                                              memory::convert_to_c(data_type)),
+                            "could not append a sum post-op");
+                else
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v3(
+                                              get(), scale,
+                                              memory::convert_to_c(data_type)),
+                                              zero_point,
+                            "could not append a sum post-op");
+        }
+        ```
+    - Pros:
+        - It is backwards compatible.
+        - Allows to use zero point without scale (scale == 1.0).
+    - Cons:
+        - It is not the simplest solution
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        attr.append_sum(scale, dt, zp);
+        ```
+
+2. Extend existing `append_sum(scale, dt)` and `dnnl_post_ops_append_sum_v2` to
+support zero point.
+    - API:
+        ```cpp
+        // include/oneapi/dnnl/dnnl.h
+        dnnl_status_t DNNL_API dnnl_post_ops_append_sum_v2(
+                dnnl_post_ops_t post_ops, float scale, int32_t zero_point,
+                dnnl_data_type_t data_type);
+
+        // include/oneapi/dnnl/dnnl.hpp
+            void append_sum(float scale = 1.f, int32_t zero_point = 0,
+                    memory::data_type data_type = memory::data_type::undef) {
+                if (data_type == memory::data_type::undef)
+                    error::wrap_c_api(dnnl_post_ops_append_sum(get(), scale),
+                            "could not append a sum post-op");
+                else
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v2(
+                                              get(), scale, zero_point,
+                                              memory::convert_to_c(data_type)),
+                            "could not append a sum post-op");
+            }
+        ```
+    - Pros:
+        - Very simple solution.
+        - Universal solution.
+    - Cons:
+        - It is not backwards compatible for C API.
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        ```
+
+3. Add possibility to use `dnnl_primitive_attr_set_zero_points` with sum.
+    - Pros:
+        - It will allow to use mask and array of zero-point values (instead of
+          one scalar value).
+    - Cons:
+        - More complicated solution.
+        - Probably requires backward incompatibility or adding a new function.
+        - Probably not needed in real topologies.
+    - Example:
+        ```cpp
+        attr.set_zero_points(DNNL_ARG_SRC_1, mask, zero_points);
+        ```
+
+### Sum zero-point design
```

Comment:
This section seems redundant with the section 1. Could you please remove it and just keep section 1?
If you still want to document the "exotic" options that are not already presented above, I would recommend to put them at the end in appendix.

Reply:
Ok, I removed this option and added information about skip zp_arg in solution.